contracts/factory/src/tests/queries.rs

```diff
@@ -221,4 +222,183 @@ fn test_deploy_multiple_liquidity_pools() {
+        admin: admin.clone(),
+        fee_recipient: user.clone(),
+        lp_wasm_hash: lp_wasm_hash.clone(),
+        max_allowed_slippage_bps: 4_000,
+        max_allowed_spread_bps: 400,
+        share_token_decimals: 6,
+        swap_fee_bps: 0,
+        token_init_info: second_token_init_info,
+        stake_init_info: second_stake_init_info,
+    };
+
+    let third_lp_init_info = LiquidityPoolInitInfo {
+        admin: admin.clone(),
+        fee_recipient: user.clone(),
+        lp_wasm_hash,
+        max_allowed_slippage_bps: 4_000,
+        max_allowed_spread_bps: 400,
+        share_token_decimals: 6,
+        swap_fee_bps: 0,
+        token_init_info: third_token_init_info,
+        stake_init_info: third_stake_init_info,
+    };
+
+    factory.create_liquidity_pool(&first_lp_init_info);
+    factory.create_liquidity_pool(&second_lp_init_info);
+    factory.create_liquidity_pool(&third_lp_init_info);
+
+    let lp_contract_addr = factory.query_pools().get(0).unwrap();
+    let first_result = factory.query_pool_details(&lp_contract_addr);
+    let share_token_addr: Address = env.invoke_contract(
+        &lp_contract_addr,
+        &Symbol::new(&env, "query_share_token_address"),
+        Vec::new(&env),
+    );
+    let first_lp_config: Config = env.invoke_contract(
+        &lp_contract_addr,
+        &Symbol::new(&env, "query_config"),
+        Vec::new(&env),
+    );
+
+    assert_eq!(
+        first_lp_init_info.max_allowed_spread_bps,
+        first_lp_config.max_allowed_spread_bps
+    );
+
+    assert_eq!(token1, first_result.pool_response.asset_a.address);
+    assert_eq!(token2, first_result.pool_response.asset_b.address);
+    assert_eq!(
+        share_token_addr,
+        first_result.pool_response.asset_lp_share.address
+    );
+    assert_eq!(lp_contract_addr, first_result.pool_address);
+
+    let second_lp_contract_addr = factory.query_pools().get(1).unwrap();
+    let second_result = factory.query_pool_details(&second_lp_contract_addr);
+    let second_share_token_addr: Address = env.invoke_contract(
+        &second_lp_contract_addr,
+        &Symbol::new(&env, "query_share_token_address"),
+        Vec::new(&env),
+    );
+    let second_lp_config: Config = env.invoke_contract(
+        &second_lp_contract_addr,
+        &Symbol::new(&env, "query_config"),
+        Vec::new(&env),
+    );
+
+    assert_eq!(
+        second_lp_init_info.max_allowed_spread_bps,
+        second_lp_config.max_allowed_spread_bps
+    );
+
+    assert_eq!(token3, second_result.pool_response.asset_a.address);
+    assert_eq!(token4, second_result.pool_response.asset_b.address);
+    assert_eq!(
+        second_share_token_addr,
+        second_result.pool_response.asset_lp_share.address
+    );
+    assert_eq!(second_lp_contract_addr, second_result.pool_address);
+
+    let third_lp_contract_addr = factory.query_pools().get(2).unwrap();
+
+    let first_lp_address_by_tuple = factory.query_for_pool_by_pair_tuple(&(token1, token2));
```

Comment:
You tested only a case with proper order of addresses. 

```suggestion
    let first_lp_address_by_tuple = factory.query_for_pool_by_pair_tuple(&(token2, token1));
```